modules/light-clients/08-wasm/types/update_test.go

```diff
@@ -594,115 +594,114 @@ func (suite *TypesTestSuite) TestUpdateState() {
+						return nil, 0, err
+					}
+
+					return &wasmvmtypes.Response{
+						Data: resp,
+					}, types.DefaultGasUsed, nil
+				})
 			},
-			true,
+			nil,
+			nil,
+		},
+		{
+			"success: client state updated on valid misbehaviour",
+			func() {
+				suite.mockVM.RegisterSudoCallback(types.UpdateStateOnMisbehaviourMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, sudoMsg []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					var msg types.SudoMsg
+					err := json.Unmarshal(sudoMsg, &msg)
+					suite.Require().NoError(err)
+
+					// set new client state in store
+					store.Set(host.ClientStateKey(), msg.UpdateStateOnMisbehaviour.ClientMessage.Data)
+					resp, err := json.Marshal(types.EmptyResult{})
+					if err != nil {
+						return nil, 0, err
+					}
+
+					return &wasmvmtypes.Response{Data: resp}, types.DefaultGasUsed, nil
+				})
+			},
+			nil,
+			[]byte("mockClientStateBz"),
+		},
+		{
+			"failure: invalid client message",
+			func() {
+				clientMsg = &tmtypes.Header{}
+				// we will not register the callback here because this test case does not reach the VM
+			},
+			fmt.Errorf("expected type %T, got %T", (*types.ClientMessage)(nil), (*tmtypes.Header)(nil)),
+			nil,
+		},
+		{
+			"failure: err return from contract vm",
+			func() {
+				suite.mockVM.RegisterSudoCallback(types.UpdateStateOnMisbehaviourMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					errMsg := errors.New("callbackfn Error")
+					return nil, 0, errMsg
+				})
+			},
+			errorsmod.Wrapf(errors.New("callbackfn Error"), "call to wasm contract failed"),
+			nil,
 		},
 	}
 
 	for _, tc := range testCases {
 		suite.Run(tc.name, func() {
-			suite.SetupWasmGrandpaWithChannel()
-			tc.setup()
+			// reset suite to create fresh application state
+			suite.SetupWasmWithMockVM()
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
 
-			if tc.expPass {
-				suite.Require().NotPanics(func() {
-					clientState.UpdateStateOnMisbehaviour(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-				})
-				clientStateBz := suite.store.Get(host.ClientStateKey())
-				suite.Require().NotEmpty(clientStateBz)
+			clientMsg = &types.ClientMessage{
+				Data: []byte("mockClientStateBz"),
+			}
+			clientState := endpoint.GetClientState()
 
-				newClientState := clienttypes.MustUnmarshalClientState(suite.chainA.Codec, clientStateBz)
-				status := newClientState.Status(suite.ctx, suite.store, suite.chainA.Codec)
-				suite.Require().Equal(exported.Frozen, status)
+			tc.malleate()
+
+			if tc.panicErr == nil {
+				clientState.UpdateStateOnMisbehaviour(suite.ctx, suite.chainA.App.AppCodec(), suite.store, clientMsg)
```

Comment:
thing we're going towards removing `suite.ctx` and `suite.store` so `suite.chainA.GetContext()` and `suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)`

Reply:
aye there's an issue already, I changed it in my PRs and if any are left they'll be caught by that issue when its time to shine comes.